app/policies/account_policy.rb

```diff
@@ -0,0 +1,18 @@
+class AccountPolicy
+  attr_reader :user, :account
+
+  def initialize(user, account)
+    raise Pundit::NotAuthorizedError, "must be a collaborator" unless user
+    @user = user
+    @account = account
+    @my_account = @user.account
+  end
+
+  def show?
+    @my_account == @account || @my_account.get_accounts.include?(@account)
```

Comment:
Maybe the `get_accounts` should be in the user.

Reply:
We can talk tomorrow, to listen your suggestion